solutions/src/main/java/ru/sberbank/school/task08/state/JAXBMapState.java

```diff
@@ -0,0 +1,57 @@
+package ru.sberbank.school.task08.state;
+
+import javax.xml.bind.annotation.XmlRootElement;
+import java.util.List;
+
+@XmlRootElement(name = "MapState")
+public class JAXBMapState<T extends GameObject> implements Savable<T> {
+    private String name;
+    private List<T> gameObjects;
+
+    private JAXBMapState() {
+        gameObjects = null;
+        name = null;
+    }
+
+    public JAXBMapState(String name, List<T> gameObjects) {
+        this.name = name;
+        this.gameObjects = gameObjects;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setGameObjects(List<T> gameObjects) {
+        this.gameObjects = gameObjects;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public List<T> getGameObjects() {
+        return this.gameObjects;
+    }
+
+    public boolean equals(final Object o) {
+        if (o == this) return true;
+        if (!(o instanceof JAXBMapState)) return false;
+        final JAXBMapState other = (JAXBMapState) o;
+        final Object this$gameObjects = this.getGameObjects();
+        final Object other$gameObjects = other.getGameObjects();
+        return this$gameObjects == null ? other$gameObjects == null : this$gameObjects.equals(other$gameObjects);
+    }
+
+    public int hashCode() {
+        final int PRIME = 59;
```

Comment:
Abbreviation in name &#39;PRIME&#39; must contain no more than &#39;2&#39; consecutive capital letters.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a461350185>*